pkgs/dsl/src/dsl.ts

```diff
@@ -210,17 +232,48 @@ export interface RuntimeOptions {
   timeout?: number;
 }
 
-// Base context interface - what ALL platforms must provide
-export interface BaseContext {
-  env: Env & ValidEnv<UserEnv>;
+// Worker configuration exposed to handlers (read-only view)
+export interface WorkerConfig {
+  maxConcurrent: number;
+  maxPollSeconds: number;
+  pollIntervalMs: number;
+  batchSize: number;
+  visibilityTimeout: number;
+}
+
+// Message record interface (minimal contract - actual type defined in @pgflow/core)
+export interface MessageRecord {
+  msg_id: number;
+  read_ct: number;
+  enqueued_at: string;
+  vt: string;
+  message: Json;
+}
+
+// Step task record interface (minimal contract - actual type defined in @pgflow/core)
+export interface StepTaskRecord<TFlow extends AnyFlow> {
+  flow_slug: string;
+  run_id: string;
+  step_slug: string;
+  input: any; // Will be properly typed by actual implementation
+  msg_id: number;
+}
+
+// Base context for queue workers (no stepTask)
+export interface BaseContext<TEnv extends Env = Env> {
+  env: TEnv & ValidEnv<UserEnv>;
```

Comment:
im not convinced that this redeclaration of UserEnv is the best approach for user-defined env vars